### PR TITLE
Support for specifying SSL certificates in Cassandra client

### LINKED ISSUE #3383: mTLS Support for Cassandra Cluster Connectivity
**Is your feature request related to a problem? Please describe.**
I'm looking to connect to a Cassandra cluster with mTLS. This requires an SSL certificate to be presented from the client-side.

**Describe the solution you'd like**
I would like to be able to specify paths to x509 encoded certificate and key files. The Cassandra client should use these files as part of its SSL configuration when communicating with the cluster.

**Describe alternatives you've considered**
A sidecar for adding the SSL encryption, but this seems a bit ridiculous given the driver supports this configuration.

**Additional context**
Sample logic showing how it is implemented - https://github.com/flightc/astra_gocql_connect/blob/master/astra_gocql_connect.go

### PR DESCRIPTION
**Pull request by bradfordcp**
**What this PR does**: Add support for Cassandra client SSL certificates

**Which issue(s) this PR fixes**:
Fixes #3383 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by bradfordcp** [comment]
Completed the checklist and squashed the commits. Waiting for the checks to pass before submitting for review.

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:38]: We're trying to uniform config. Could you rename the YAML config options and their CLI flags respectively to:

- `tls_cert_path` / `tls-cert-path`
- `tls_key_path` / `tls-key-path`

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:67]: `certifcate` > `certificate`

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:94]:
```suggestion
		return errors.Errorf("TLS certificate specified, but private key configuration is missing.")
```

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:97]:
```suggestion
		return errors.Errorf("TLS private key specified, but certificate configuration is missing.")
```

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:159]:
```suggestion
		if cfg.CertPath != "" {
```

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:162]:
```suggestion
				return errors.Wrap(err, "unable to load TLS certificate and private key")
```

File: pkg/chunk/cassandra/storage_client_test.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client_test.go:98]: Could you assert on the exact error message, please?

_Same comment applies to the next test._

**Review by pracucci — COMMENTED** [review]
Good job, thanks! I left few minor comments and then we should be good to go.

**Comment by bradfordcp** [comment]
I'll spend some cycles on getting this turned around tonight. Thanks for the review!

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:94]:
```suggestion
		return errors.Errorf("TLS certificate specified, but private key configuration is missing.")
```

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:68]:
```suggestion
	f.StringVar(&cfg.KeyPath, "cassandra.tls-key-path", "", "Path to private key file used by TLS.")
```

File: pkg/chunk/cassandra/storage_client.go
Comment (pracucci) [inline pkg/chunk/cassandra/storage_client.go:67]:
```suggestion
	f.StringVar(&cfg.CertPath, "cassandra.tls-cert-path", "", "Path to certificate file used by TLS.")
```

**Review by pracucci — APPROVED** [review]
LGTM, thanks! I left a couple of nits and we're good to go. Thanks for your patience!

**Comment by bradfordcp** [comment]
Thanks for your patience, I'm surprised by the number of typos :sweat_smile:

**Comment by pracucci** [comment]
> Thanks for your patience, I'm surprised by the number of typos 😅

Thanks to you @bradfordcp. LGTM. Let me ping another maintainer for the 2nd review and then we can merge it.

**Review by jtlisi — APPROVED** [review]
LGTM

Thank you for the PR!

**Pull request merged**
